### ISSUE TITLE
Support for FosUserBundle 2.X

### ISSUE THREAD

**Issue by rande** [sonata-project/SonataUserBundle#322]
This issue regroups information about support FosUserBundle 1.x and 2.x.

**Comment by numerogeek** [comment]
+1

**Comment by limfinity** [comment]
+1

**Comment by Tetdoss** [comment]
+1

**Comment by lumilo8** [comment]
We have the same problem... integrating https://github.com/PUGX/PUGXMultiUserBundle it requires fosuser 2.0 :(
:+1:

**Comment by VolodymyrChornohal** [comment]
@rande hi! what version or release of SUB should I try in order to satisfy FOSUB2.0 composer requirements? I've checked all the releases and tagged versions but the requirement is always FOSUB1.3 :(

**Comment by extdevelopment** [comment]
+1

**Comment by OskarStark** [comment]
:+1:

**Comment by greg0ire** [comment]
Please stop the "+1", "me too" comments.

**Comment by rande** [comment]
@greg0ire a PR is more efficient ;)

**Comment by greg0ire** [comment]
Yeah definitely :) Also, it would be great if github had a built-in +1 feature so that people could vote for issues without leaving the comment and notifying everyone in the thread. @everyone : Please **do not** +1 that comment if you agree.

**Comment by pulse00** [comment]
I'm working on a PR for [`fos-user-bundle/2.x`](https://github.com/pulse00/SonataUserBundle/tree/feature/fos-2.x) support. 

I have a question regarding the idea behind naming the controllers `**FOSUser1Controller`. I've currently simply added a second set of controllers/routings using `**FOSUser2Controller` which work with the fos-user 2.x version. Do i understand it right that both 1.x and 2.x versions of the FUB should be supported?

What about the version constraint in `composer.json`, should it be set to something like `>=1.0,<=2.0` ?

**Comment by phansys** [comment]
IMO, ```"friendsofsymfony/user-bundle": "~2.0"``` would be great for FOSUB 2.x support.

**Comment by pulse00** [comment]
@rande any input on this would be great.

**Comment by roman-1983** [comment]
+1

**Comment by rande** [comment]
@pulse00 sorry, I didn't see your comment ;)

**Comment by rande** [comment]
@pulse00 the FosUserBundle 2.0 is not yet stable, but people is using it. So yes I would like to support both version.

**Comment by pulse00** [comment]
@rande alright, thanks for the info

**Comment by numerogeek** [comment]
@rande @pulse00 hi guys, any news about this issue ? 
There is a compatibility conflict if we also want to use PugxMultiUserBundle which require FOSUB 2.0.
Is there any way to use SonataAdmin without SonataUserBundle ? 

Thanks

**Comment by klairemichel** [comment]
+1

**Comment by VolodymyrChornohal** [comment]
@pulse00 what work exactly should be done on the controllers? could you share your progress?

**Comment by pulse00** [comment]
@VolodymyrChornohal the main difference in fos 2.x is that the controllers now use events to let you hook into their logik. Currently i have a working solution which can be found here https://github.com/sonata-project/SonataUserBundle/compare/master...pulse00:feature/fos-2.x . 

But it need's to be cleaned up before i can send a PR.

I won't be able to finish this until next week, so if anyone wants to help i can give them write access to the fork.

**Comment by DavidSoussan** [comment]
@pulse00 I would be happy to help clean up the code, been waiting for this as we need user events. How do we do this then?

**Comment by pulse00** [comment]
@DavidSoussan i've created a ticket in the fork, so we don't pollute this thread here https://github.com/pulse00/SonataUserBundle/issues/1

**Comment by VolodymyrChornohal** [comment]
@pulse00 awesome , thanks, i'd also do :+1:

**Comment by justi** [comment]
+1

**Comment by JHGitty** [comment]
+1 :+1:

**Comment by chartinger** [comment]
+1

**Comment by gyrocopter** [comment]
+1

**Comment by redactuk** [comment]
+1

**Comment by JHGitty** [comment]
This feature is needed for Symfony 3 support. (#647)

**Comment by paquitodev** [comment]
@rande I am starting a new project and I would like to use Symfony 2.8 but this bundle isn't compatible with FOSUser 2.x so +1 for this, could you give us any advice around this?

**Comment by rande** [comment]
seems FosUserBundle 2.x will only support Symfony3. @pulse00 are you still want to port your work to work with sf2.8 and 3.0 only ? This will save me a lot of time ;)

Thanks

**Comment by OskarStark** [comment]
any news here?

**Comment by yannskee** [comment]
apparently this could be a fix : 
"friendsofsymfony/user-bundle": "2.0.x-dev as 1.3.2"

**Comment by rande** [comment]
For information, the branch https://github.com/sonata-project/SonataUserBundle/tree/add_support_for_fos_user2 will add support for FosUser2. The work is based on @pulse00's work. Please feel free to send PR to fix latest issues, this will help to be merged. Thanks

**Comment by murnieza** [comment]
Thanks @yannskee . This at least lets me install the bundle. I'll see how it works.

**Comment by redactuk** [comment]
The solution here is the commuity put some pressure on FOSUserBundle maintainers to get on with making a 2.x release. The fact v2 has been Alpha since Sep 14 is quite unbelievable.

**Comment by ayalon** [comment]
@rande : I would like to help to get this branch merged.

I was adding these lines to my composer.json:
`"sonata-project/user-bundle": "dev-add_support_for_fos_user2",`
`"friendsofsymfony/user-bundle": "2.0.x-dev",`

Unfortunatly I have an error, that I cannot resolve from then on:

`PHP Fatal error:  Uncaught exception 'Symfony\Component\DependencyInjection\Exception\ServiceNotFoundException' with message 'The service "sonata.user.user_manager" has a dependency on a non-existent service "fos_user.entity_manager`.

Did I miss something? From your point of view the branch should be testable and I don't know how to fix that.

**Comment by murnieza** [comment]
@ayalon you want to add alias as service was renamed:

```yml
# app/config/services.yml
services:
    fos_user.entity_manager:
        alias: fos_user.object_manager
```

**Comment by redactuk** [comment]
With reference this thread https://github.com/FriendsOfSymfony/FOSUserBundle/issues/2067 has Sonata team ever considered a permanenk fork of FOSUserBundle perhaps merging with SonataUserBundle?

**Comment by kilhage** [comment]
+1

**Comment by cosf87** [comment]
+1

**Comment by vazgen** [comment]
+1

**Comment by greg0ire** [comment]
There is a +1 button now, you know.

**Comment by greg0ire** [comment]
Please go test / provide feeback on #745

**Comment by responseconcepts** [comment]
I've managed to make this combination finally working for Symfony 3:
        "friendsofsymfony/user-bundle": "dev-master#d1fccc6335b74beca94525551cacc06b256ed261",
        "sonata-project/admin-bundle": "^3.1",
        "sonata-project/core-bundle": "^3.0",
        "sonata-project/datagrid-bundle": "^2.2",
        "sonata-project/doctrine-orm-admin-bundle": "^3.0",
        "sonata-project/user-bundle": "dev-rc_sf3tweaks",
        "sonata-project/formatter-bundle": "^3.0"

using the forked SonataUserBundle and creating branch rc_sf3tweaks

 "repositories": [
        {
            "type": "vcs",
            "url": "https://github.com/responseconcepts/SonataUserBundle"
        }
    ]

There are some tweaks and hacks, but looks like it's working, If anyone interested to improve it, please take a look: https://github.com/responseconcepts/SonataUserBundle/commits/rc_sf3tweaks

**Comment by ZajdeR** [comment]
It don't work for me.

**Comment by responseconcepts** [comment]
@ZajdeR Can you describe more what is not working? Do you use the same set of sonata libraries and Doctrine ORM ?

**Comment by ZajdeR** [comment]
Maybe I'm ding sonthing wrong.
       "friendsofsymfony/user-bundle": "dev-master#d1fccc6335b74beca94525551cacc06b256ed261",

        "sonata-project/admin-bundle": "^3.1",
        "sonata-project/doctrine-orm-admin-bundle": "^3.0",
        "sonata-project/easy-extends-bundle": "3.x-dev",
        "sonata-project/user-bundle": "dev-rc_sf3tweaks",
        
        "sonata-project/core-bundle": "^3.0",
        "sonata-project/datagrid-bundle": "^2.2",
        "sonata-project/formatter-bundle": "^3.0"


Problem 1
    - The requested package sonata-project/user-bundle dev-rc_sf3tweaks exists as sonata-project/user-bundle[2.0.0, 2.1.0, 2.1.1, 2.2.0, 2.2.1, 2.2.2, 2.2.3, 2.2.4, 2.2.5, 2.x-dev, 3.0.0, 3.0.1, 3.x-dev, dev-add_support_for_fos_user2, dev-develop, dev-fix_deprecation_notices, dev-master, 4.x-dev] but these are rejected by your constraint.

**Comment by responseconcepts** [comment]
Did you add my repository to your composer.json

` "repositories": [
        {
            "type": "vcs",
            "url": "https://github.com/responseconcepts/SonataUserBundle"
        }
    ]`

**Comment by ZajdeR** [comment]
I didn't, ok now works
but I have to also change  

    "sonata-project/easy-extends-bundle": "3.x-dev",

to

    "sonata-project/easy-extends-bundle": "2.x-dev",

thanks for help

**Comment by Soullivaneuh** [comment]
@ZajdeR You can also use a stable version for easy-extends: https://packagist.org/packages/sonata-project/easy-extends-bundle#2.1.10

**Comment by progmax** [comment]
It works at the moment?

**Comment by core23** [comment]
Please stop asking until we got a stable release for the FOSUserBundle or use the fos2 branch

**Comment by AlexandreKilian** [comment]
I understand you'd want to wait for a stable fos2, but fos1.X isn't compatible with symfony3.x, so right now, there is no way of using any of it right now... or am I missing something here?

**Comment by greg0ire** [comment]
You can probably inline alias fos2 to fos1 to be able to install it. BTW, [looks like there is going to be a beta release](https://github.com/FriendsOfSymfony/FOSUserBundle/pull/2329) (instead of an alpha release)

**Comment by jontorrado** [comment]
@greg0ire the beta 1 was released 4 days ago (https://github.com/FriendsOfSymfony/FOSUserBundle/releases). Maybe it's time to start creating a "beta SonataUserBundle version" too. I'll be glad to help.

**Comment by greg0ire** [comment]
Isn't it what [this branch](https://github.com/sonata-project/SonataUserBundle/tree/add_support_for_fos_user2) is about?

**Comment by jontorrado** [comment]
Last commit 27 of July.

**Comment by greg0ire** [comment]
So? Isn't it better than nothing? Would you rather start from scratch? Honest question, didn't look what is in this branch.

**Comment by jontorrado** [comment]
I think I really would xD

**Comment by greg0ire** [comment]
OK… you can start something if you want, and make a PR on 3.x if you want so that we can discuss it. We won't merge it because we only rely on stable software in the 3.x and master branches, but we might create a new branch for people who want to prepare SonataUserBundle for the next FOSUserBundle major release.

**Comment by AmaliDeveloper** [comment]
you can get th solution in this tutorial [link](https://youtu.be/KT2JHTKziPo)

**Comment by jontorrado** [comment]
So, finally, 2.0.0 has been realeased. Let me know if I can help with this ;) https://github.com/FriendsOfSymfony/FOSUserBundle/releases/tag/v2.0.0

**Comment by jordisala1991** [comment]
There is a branch on this bundle: https://github.com/sonata-project/SonataUserBundle/tree/add_support_for_fos_user2

Help testing it and improving it.

Also there is a PR opened: https://github.com/sonata-project/SonataUserBundle/pull/861
You could help reviewing it and testing it.

:)

**Comment by jordisala1991** [comment]
Or there is another one that is trying to add compatibility with 1.3.x and 2.0.0 in 3.x branch: https://github.com/sonata-project/SonataUserBundle/pull/745

Doing this on 3.x will help getting 2.0.0 supported, because is the stable branch

**Comment by OskarStark** [comment]
IMO we should NOT support both version, as we discovered that code changed a lot between both version, maintaining both in on version could be very painful in the future

**Comment by jordisala1991** [comment]
I think the same, but that will require more work to get master ready to release 4.0 also.

**Comment by rusanter** [comment]
@jordisala1991, why only php7 is supported in branch add_support_for_fos_user2?
https://github.com/sonata-project/SonataUserBundle/blob/add_support_for_fos_user2/composer.json#L20

**Comment by jordisala1991** [comment]
If we do it for master not BC (not on the 3.x branch), it will require php 7 at some point because:

https://github.com/sonata-project/dev-kit/pull/225

We should prepare on this `add_support_for_fos_user2` branch for this. WDYT @sonata-project/contributors ?

**Comment by core23** [comment]
👍 👍 👍 for dropping PHP 5

**Comment by jontorrado** [comment]
@rusanter if you are still using PHP 5.*, you are doing something wrong (IMHO). Also, not having to check for PHP 5.* breaks will make this development faster.

**Comment by leylaJudy** [comment]
guess i'll gonna drop using sonataAdmin just  because of this issue

**Comment by jordisala1991** [comment]
There is an open PR already to support both SF3 and FOSUSer 2.0: https://github.com/sonata-project/SonataUserBundle/pull/869

**Comment by OskarStark** [comment]
Closed via #869
